### PR TITLE
Codebase improvements

### PR DESCRIPTION
**Pull request by StackOverflowExcept1on**
Changes:
- [x] add `Into<T>` approach to reduce the number of `impl`s
- [x] replace `()` with `_` (this is better for future codegen, `grep -r --include="*.rs" "self.fields" | grep "()"`)
- [x] refactoring of marker traits
- [x] typo fixes
- [x] add `for_reply()` method to the `SendBuilder`

### CODE REVIEW

File: src/common.rs [hunk 1]
```diff
@@ -15,19 +15,27 @@ pub struct DelayW(pub(crate) u32);
...
 // Currently the `GasLimit` and `ReservationId` generics do not overlap,
 // but in future versions of gstd they may be activated at the same time.
 // https://github.com/gear-tech/gear/pull/2705
 
-pub trait GasLimitUnitMarker {} //may be removed soon
-
-impl GasLimitUnitMarker for () {}
-
-pub trait ReservationIdUnitMarker {} //may be removed soon
+pub trait UnitTypeMarker {} //may be removed soon
```
Comment (breathx) [inline src/common.rs]:
```suggestion
//
// To be removed soon
pub trait UnitTypeMarker {}
```

**Review by breathx — APPROVED** [review]
Rename the pr to be more descriptive.
And would you simplify other fields with `Into` approach?

File: src/lib.rs [hunk 2]
```diff
@@ -1,6 +1,12 @@
 #![no_std]
 
+#[allow(unused_imports)]
```
Comment (breathx) [inline src/lib.rs:3]:
```suggestion
```

**Review by breathx — APPROVED** [review]
Never force-push already reviewed PRs. And still consider new title for PR

File: src/lib.rs (hunk 2 above)
Comment (StackOverflowExcept1on) [reply src/lib.rs:3]: this is fix for a failed CI (it fails the `cargo clippy` check)

File: Cargo.toml [hunk 3]
```diff
@@ -4,10 +4,15 @@ version = "0.1.0"
...
-repository = "https://github.com/StackOverflowExcept1on/gstd-fluent"
-license = "GPL-3.0"
+repository = "https://github.com/gear-foundation/gstd-fluent"
+license = "MIT"
 keywords = ["gstd", "smart-contracts", "gear-tech", "gear-foundation"]
 categories = ["api-bindings", "no-std"]
 
 [dependencies]
-gstd = { git = "https://github.com/gear-tech/gear.git", branch = "av/gstd-panic-handler", default-features = false }
+gstd = { git = "https://github.com/gear-tech/gear.git", rev = "f4407fa" }
```
Comment (shamilsan) [inline Cargo.toml:13]: Could we use `946ac47` rev as all other contracts are at it?
```suggestion
gstd = { git = "https://github.com/gear-tech/gear.git", rev = "946ac47" }
```

**Review by shamilsan — APPROVED** [review]

**Pull request merged**